testing_local_changes.md

```diff
@@ -0,0 +1,115 @@
+
+When using OpenShift Local (aka CRC), your export will be something like this:
+
+```
+% cat &lt;&lt;EOF &gt;/etc/exports
+${HOME}/edpm-ansible 192.168.130.0/24(rw,sync,no_root_squash)
+EOF
+
+% exportfs -r
+```
+
+**NOTE:** CRC installs its own firewall rules, which likely will need to be
+adjusted depending on the location of your NFS server. If your edpm-ansible
+directory is on the same system that hosts your CRC, then the simplest thing
+to do is insert a rule that essentially circumvents the other rules:
+
+```
+% nft add rule inet firewalld filter_IN_libvirt_pre accept
+```
+
+# Create edpm-ansible PV and PVC
+
+Create an NFS PV, and a PVC that can be mounted on the ansibleee pods.
+
+**NOTE:** While it's possible to add an NFS volume directly to a pod, the
+default k8s Security Context Constraint (SCC) for non-privileged pods does
+not permit NFS volume mounts. The approach of using an NFS PV and PVC works
+just as well, and avoids the need to fiddle with SCC policies.
+
+```
+% # E.g. ${HOME}/edpm-ansible
+% NFS_SHARE=&lt;Path to your edpm-ansible directory&gt;
+% NFS_SERVER=&lt;IP of your NFS server&gt
+% cat &lt;&lt;EOF &gt;edpm-ansible-storage.yaml
+apiVersion: v1
+kind: PersistentVolume
+metadata:
+  name: edpm-ansible
+spec:
+  capacity:
+    storage: 1Gi
+  volumeMode: Filesystem
+  accessModes:
+    - ReadOnlyMany
+  # IMPORTANT! The persistentVolumeReclaimPolicy must be "Retain" or else
+  # your code will be deleted when the volume is reclaimed!
+  persistentVolumeReclaimPolicy: Retain
+  storageClassName: edpm-ansible
+  mountOptions:
+    - hard
+    - nfsvers=4.1
+  nfs:
+    path: ${NFS_SHARE}
+    server: ${NFS_SERVER}
+---
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: edpm-ansible
+spec:
+  storageClassName: edpm-ansible
+  accessModes:
+    - ReadOnlyMany
+  resources:
+    requests:
+      storage: 1Gi
+EOF
+
+% oc apply -f edpm-ansible-storage.yaml
+```
+
+# Add extraMount to your OpenStackDataPlane CR
+
+Use kustomize or "oc edit" to add the edpm-ansible PVC to the
+OpenStackDataPlane's /spec/roles/edpm-compute/nodeTemplate/extraMounts. The
+mountPath is where the edpm-ansible *roles* and *plugins* directories are
+located inside the openstack-ansibleee-runner container image. The
+OpenStackDataPlane CR should contain the following snippet:
+
+```
+  spec:
+    roles:
+      edpm-compute:
+        nodeTemplate:
+          extraMounts:
+          - extraVolType: edpm-ansible
+            mounts:
+            - mountPath: /usr/share/ansible/collections/ansible_collections/osp/edpm
+              name: edpm-ansible
+            volumes:
+            - name: edpm-ansible
+              persistentVolumeClaim:
+                claimName: edpm-ansible
+                readOnly: true
+```
```

Comment:
ack so with this extra mount deffitnion you have effectly replace the ansibel collection in the AEE pod that is spawned win a role invokes a OpenStackDatapalne service.

it will instead use the source form the nfs share via the pvc.

Reply:
Yes, the ansibleee pod will use your local edpm-ansible sources.